tests/helics/application_api/CMakeLists.txt

```diff
@@ -7,57 +7,91 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-set(application_api_test_headers testFixtures.hpp ValueFederateTestTemplates.hpp
-                                 ../coreTypeLists.hpp
-)
+set(common_test_headers testFixtures.hpp ../coreTypeLists.hpp)
 
-set(application_api_test_sources
-    MessageFederateKeyTests.cpp
+set(common_test_sources testFixtures.cpp applicationApiTests.cpp)
+
+set(valueFederate_test_sources
     ValueFederateKeyTests.cpp
-    FilterTests.cpp
     TranslatorTests.cpp
-    testFixtures.cpp
+    ValueFederateTestTemplates.hpp
     PrimaryTypeConversionTests.cpp
-    CallbackFederateTests.cpp
-    FilterAdditionalTests.cpp
     ValueConverterTests.cpp
-    application-api-tests.cpp
     data_viewTests.cpp
-    MessageFederateAdditionalTests.cpp
-    FederateTests.cpp
     subPubObjectTests.cpp
```

Comment:
These seem like they are a better fit for the application api tests, rather than value federate (depending on what the subPubObjectTests.cpp are doing).

Reply:
The PrimaryTypeConversion, ValueConverterTests, and subPubObjectTests are inherent parts of the value Federate data translation layers, so I think they belong in there,  dataView could be be too but I might move that to the applicationApi since that isn't completely specific to the valueFederate